compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaXMLTypeSymbol.java

```diff
@@ -60,7 +60,9 @@ public BallerinaXMLTypeSymbol(CompilerContext context, ModuleID moduleID, BXMLTy
             if (xmlType == symbolTable.xmlType || this.typeParameter().isEmpty()) {
                 this.typeName = "xml";
             } else {
-                this.typeName = "xml<" + this.typeParameter().get().getName().orElse("") + ">";
+                // if type param name is missing, it should default to xml<never>
+                String typeParamString = this.typeParameter().get().getName().orElse("never");
```

Comment:
@rdhananjaya @suleka96 Can you check this? I think this is not correct.

Reply:
@pubudu91 Can you check this.

Reply:
While the spec doesn't seem to mention this case, I also don't think this behaviour is implied. In the implementation, we have considered `xml` type as the union of `Element`, `Text`, `ProcessingInstruction` and `Comment`. And I think that's the implied behaviour of the spec as well.

Reply:
The issue is with `xml<never>` being shown as `xml<>` from the signature, and `xml<>` being syntactically incorrect. #28693 
And I don't think assuming `xml<never>` as same as `xml` is correct either, since following would be possible then, no?

```ballerina
public function main(){
    var g = <xml<never>> xml``;
    g = xml `<a></a>`;
}
```

Reply:
Shall we change this like the following? 
- Move the logic in `getName()` to `signature()`. In `getName()`, let's just return `xml` for now
- Then, call `String typeParamString = this.typeParameter().get().signature();`

Reply:
Returning `xml` in `getName()` caused some other issues, so I refactored that code to the `signature` and did this change there